### PR TITLE
CDPS-592: Don't display recognised listener unless yes when given no suitible

### PR DESCRIPTION
**Pull request by danielburnley**
Updated based on the logic listed here: 
https://dsdmoj.atlassian.net/browse/NN-2579

### CODE REVIEW

**Review by davidmiddletonmoj — APPROVED** [review]

**Comment by whitfield-mj** [comment]
Are there no tests for this logic? Feels like there should be tests

File: server/services/overviewPageService.ts [hunk 1]
```diff
@@ -593,7 +593,14 @@ export default class OverviewPageService {
...
 
-    if ((recognised === BooleanString.Yes || recognised === BooleanString.No) && suitable !== BooleanString.No) {
+    if (!suitable && recognised === BooleanString.Yes) {
+      statusList.push({
+        label: 'Recognised Listener',
+      })
+    } else if (
+      (recognised === BooleanString.Yes || recognised === BooleanString.No) &&
+      suitable === BooleanString.Yes
+    ) {
```
Comment (whitfield-mj) [inline server/services/overviewPageService.ts:603]: It's blowing my mind. Why would recognised === 'No' mean that it is recognised?!

**Comment by davidmiddletonmoj** [comment]
> Are there no tests for this logic? Feels like there should be tests

There were already tests for everything but the combo that failed. Dan has added that in along with the new logic.

It does seem odd, but that is what is in the specs

**Comment by danielburnley** [comment]
I've updated the original description with the link to where the original logic is, it is very unexpected logic

**Pull request merged**